### PR TITLE
Introduce `WeightToFee` trait instead of `WeightToFeePolynomial` and make `WeightToFeePolynomial` implement it instead

### PR DESCRIPTION
**Pull request by nazar-pc**
Fixes https://github.com/paritytech/substrate/issues/11278

This removes `WeightToFee` and `LengthToFee` constants from `pallet-transaction-payment` as suggested in https://github.com/paritytech/substrate/pull/11226#issuecomment-1108321002 by @kianenigma (@jacogr this may require your attention for UI updates).

Polkadot companion: https://github.com/paritytech/polkadot/pull/5525

Polkadot address: 1vSxzbyz2cJREAuVWjhXUT1ds8vBzoxn2w4asNpusQKwjJd

### CODE REVIEW

**Comment by jacogr** [comment]
Have an implementation in the apps UI, but it is/was actually completely unused everywhere ...anyway removed [here now](https://github.com/polkadot-js/apps/pull/7581) seems to be stuff from pre paymentQueryInfo (so it was a worth-while check). 

However, one thing to just check in on is sidecar, I am not sure about the in-use status, but there is [some code here](https://github.com/paritytech/substrate-api-sidecar/blob/847355b43e2caca79a8134b9b47fad814b4dfebf/src/services/blocks/BlocksService.ts#L526-L535)

File: frame/support/src/weights.rs [hunk 1]
```diff
@@ -650,6 +650,15 @@ pub struct WeightToFeeCoefficient<Balance> {
 /// A list of coefficients that represent one polynomial.
 pub type WeightToFeeCoefficients<T> = SmallVec<[WeightToFeeCoefficient<T>; 4]>;
 
+/// A trait that describes the weight to fee calculation.
+pub trait WeightToFee {
+	/// The type that is returned as result from calculation.
+	type Balance: BaseArithmetic + From<u32> + Copy + Unsigned;
+
+	/// Calculates the fee from the passed `weight`.
+	fn calc(weight: &Weight) -> Self::Balance;
```
Comment (kianenigma) [inline frame/support/src/weights.rs:659]:
```suggestion
	/// Calculates the fee from the passed `weight`.
	fn weight_to_fee(weight: &Weight) -> Self::Balance;
```

opinionated, but I think `pub trait SomeName { fn some_name() {} }` is an acceptable idiom. Sticking to `.calc` seems like restricting yourself based on the previous `WeightToFeePolynomial`

File: frame/transaction-payment/asset-tx-payment/src/tests.rs [hunk 2]
```diff
@@ -130,23 +125,27 @@ impl pallet_balances::Config for Runtime {
...
 	type Balance = u64;
 
-	fn polynomial() -> WeightToFeeCoefficients<Self::Balance> {
-		smallvec![WeightToFeeCoefficient {
-			degree: 1,
-			coeff_frac: Perbill::zero(),
-			coeff_integer: WEIGHT_TO_FEE.with(|v| *v.borrow()),
-			negative: false,
-		}]
+	fn wight_to_fee(weight: &Weight) -> Self::Balance {
```
Comment (Garandor) [inline frame/transaction-payment/asset-tx-payment/src/tests.rs:139]:
```suggestion
	fn weight_to_fee(weight: &Weight) -> Self::Balance {
```
along with its 26 uses

Comment (nazar-pc) [reply frame/transaction-payment/asset-tx-payment/src/tests.rs]: :facepalm:

**Review by shawntabrizi — APPROVED** [review]
looks good to me

**Comment by shawntabrizi** [comment]
companion needs fixes

**Comment by nazar-pc** [comment]
`cargo check --all-targets` compiles for me locally without issues, not sure what I did wrong with companion

**Comment by shawntabrizi** [comment]
> `cargo check --all-targets` on Polkadot branch compiles for me locally without issues, not sure what I did wrong with companion

After doing deiner?

**Comment by shawntabrizi** [comment]
You can see the errors in CI:

```
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
100 |     use frame_support::weights::WeightToFee;
```

**Comment by nazar-pc** [comment]
I see now, pushed fix for those

**Comment by shawntabrizi** [comment]
looks good now

**Review by kianenigma — APPROVED** [review]

**Comment by shawntabrizi** [comment]
bot merge

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**